torch/ao/quantization/fx/_decomposed.py

```diff
@@ -0,0 +1,52 @@
+import torch
+from torch.library import Library, impl
+
+# Note: decomposed means decomposed quantized tensor, using decomposed so that the
+# name is not too long
+quantized_decomposed_lib = Library("quantized_decomposed", "DEF")
+
+quantized_decomposed_lib.define(
+    "quantize_per_tensor(Tensor input, float scale, int zero_point, int quant_min, int quant_max, ScalarType dtype) -> Tensor")
+
+@impl(quantized_decomposed_lib, "quantize_per_tensor", "CompositeExplicitAutograd")
+def quantize_per_tensor(input, scale, zero_point, quant_min, quant_max, dtype):
+    assert input.dtype == torch.float32, f"Expecting input to have dtype torch.float32, but got dtype: {input.dtype}"
+    quant_min_lower_bound = 0
+    quant_max_upper_bound = 0
+    if dtype == torch.uint8:
+        quant_min_lower_bound = 0
+        quant_max_upper_bound = 255
+    elif dtype == torch.int8:
+        quant_min_lower_bound = -128
+        quant_max_upper_bound = 127
+    else:
+        raise ValueError(f"Unsupported dtype: {dtype}")
+
+    assert quant_min >= quant_min_lower_bound, \
+        "quant_min out of bound for dtype, " \
+        f"quant_min_lower_bound: {quant_min_lower_bound} quant_min: {quant_min}"
+
+    assert quant_max <= quant_max_upper_bound, \
+        "quant_max out of bound for dtype, " \
+        f"quant_max_upper_bound: {quant_max_upper_bound} quant_max: {quant_max}"
+
+    inv_scale = 1.0 / scale
+    return torch.clamp(torch.round(input * inv_scale) + zero_point, quant_min, quant_max).to(dtype)
+
+# Note: quant_min/quant_max/dtype are not used in the operator, but for now it's kept in
+# the signature as metadata for the input Tensor, this might be useful for pattern
+# matching in the future
+# We will revisit this later if we found there are no use cases for it
+quantized_decomposed_lib.define(
+    "dequantize_per_tensor(Tensor input, float scale, int zero_point, int quant_min, int quant_max, ScalarType dtype) -> Tensor")
+
+@impl(quantized_decomposed_lib, "dequantize_per_tensor", "CompositeExplicitAutograd")
+def dequantize_per_tensor(input, scale, zero_point, quant_min, quant_max, dtype):
+    assert input.dtype == dtype, f"Expecting input to have dtype: {dtype}"
+    if dtype in [torch.uint8, torch.int8]:
+        # TODO: investigate why
+        # (input - zero_point).to(torch.float32) * scale
```

Comment:
nit: I think this fails because of the under-/overflow? For example, if `dtype(input) == uint8`, `input == 0`, and `zero_point > 0`, you will have wrapped around values:

```
>>> import torch
>>> a = torch.tensor([1,2,3], dtype=torch.uint8)
>>> a - 3
tensor([254, 255,   0], dtype=torch.uint8)
```

Reply:
yeah could be, but I remember seeing fbgemm is doing the same: https://github.com/pytorch/FBGEMM/blob/main/include/fbgemm/QuantUtils.h#L139-L141, or maybe the subtraction in `(src - qparams.zero_point)` is an int32 since qparams.zero_point is int32: https://github.com/pytorch/FBGEMM/blob/main/include/fbgemm/QuantUtilsAvx2.h#L21